yasa/plotting.py

```diff
@@ -235,34 +227,38 @@ def plot_spectrogram(
         >>> # Load the 30-sec hypnogram and upsample to data
         >>> hypno = np.loadtxt('https://raw.githubusercontent.com/raphaelvallat/yasa/master/notebooks/data_full_6hrs_100Hz_hypno_30s.txt')
         >>> hypno = yasa.hypno_upsample_to_data(hypno, 1/30, data, sf)
-        >>> fig = yasa.plot_spectrogram(data, sf, hypno, cmap='Spectral_r')
+        >>> hypno = yasa.hypno_int_to_str(hypno)
+        >>> hyp = yasa.Hypnogram(hypno, freq="10ms")
+        >>> fig = yasa.plot_spectrogram(data, sf, hyp, cmap='Spectral_r')
     """
     # Increase font size while preserving original
     old_fontsize = plt.rcParams["font.size"]
     plt.rcParams.update({"font.size": 18})
 
     # Safety checks
-    assert isinstance(data, np.ndarray), "Data must be a 1D NumPy array."
-    assert isinstance(sf, (int, float)), "sf must be int or float."
-    assert data.ndim == 1, "Data must be a 1D (single-channel) NumPy array."
-    assert isinstance(win_sec, (int, float)), "win_sec must be int or float."
-    assert isinstance(fmin, (int, float)), "fmin must be int or float."
-    assert isinstance(fmax, (int, float)), "fmax must be int or float."
-    assert fmin < fmax, "fmin must be strictly inferior to fmax."
-    assert fmax < sf / 2, "fmax must be less than Nyquist (sf / 2)."
-    assert isinstance(vmin, (int, float, type(None))), "vmin must be int, float, or None."
-    assert isinstance(vmax, (int, float, type(None))), "vmax must be int, float, or None."
+    assert isinstance(data, np.ndarray), "`data` must be a 1D NumPy array."
+    assert isinstance(sf, (int, float)), "`sf` must be int or float."
+    assert data.ndim == 1, "`data` must be a 1D (single-channel) NumPy array."
+    assert isinstance(win_sec, (int, float)), "`win_sec` must be int or float."
+    assert isinstance(fmin, (int, float)), "`fmin` must be int or float."
+    assert isinstance(fmax, (int, float)), "`fmax` must be int or float."
+    assert fmin < fmax, "`fmin` must be strictly inferior to `fmax`."
+    assert fmax < sf / 2, "`fmax` must be less than Nyquist (sf / 2)."
+    assert isinstance(vmin, (int, float, type(None))), "`vmin` must be int, float, or None."
+    assert isinstance(vmax, (int, float, type(None))), "`vmax` must be int, float, or None."
     if vmin is not None:
-        assert isinstance(vmax, (int, float)), "vmax must be int or float if vmin is provided"
+        assert isinstance(vmax, (int, float)), "`vmax` must be int or float if `vmin` is provided."
     if vmax is not None:
-        assert isinstance(vmin, (int, float)), "vmin must be int or float if vmax is provided"
-    if hypno is not None:
-        # Validate and handle hypnogram-related inputs
-        assert hypno.size == data.size, "Hypno must have the same sf as data."
+        assert isinstance(vmin, (int, float)), "`vmin` must be int or float if `vmax` is provided."
+    if hyp is not None:
+        from yasa.hypno import Hypnogram  # Avoiding circular import
```

Comment:
Just curious: can you say more about this? Importing Hypnogram at the top of the file would result in a circular import?

Reply:
At the top of the `hypno` module: `from yasa.plotting import plot_hypnogram`
At the top of the `plotting` module: `from yasa.hypno import Hypnogram`

This would result in a infinite loop of imports and so you get the circular import error. I don't have a lot of experience with this, so there might be a more standard solution than within-function imports. Let me know if you have another idea!

Reply:
also, in some cases it's only imported for assertion check and could just be left out

Reply:
Hmm, I've never really worried about that but you're absolutely right. Within-function imports sound like a good strategy